cache properties by tag in xml-properties-mixin to improve performance

chadkirby opened the pull request:
When validating an 85 page document, we were spending >10-12 seconds traversing the dom to find properties. This PR speeds up the overall validation process by ~8%.

chadkirby committed:
cache properties by tag in xml-properties-mixin to improve performance slightly

When validating an 85 page document, we were spending >10-12 seconds traversing the dom to find properties. This commit speeds up the overall validation process by ~8%.

bendemboski commented on src/xml-properties-mixin.js:
```diff
@@ -61,8 +61,9 @@ let XmlPropertiesMixin = (superclass) => class extends superclass {
-      this._Pr = $Pr;
+      // once our $Pr has been created, it shouldn't change, so we'll replace
+      // this getter with a simple property
+      Object.defineProperty(this, `$Pr`, { value: $Pr });
```
Can you leave some kind of comment about the expected lifetime of objects using this mixin or something? Just reading the code, it looks like if you ever call `removePr()` you perma-bork your object because the removed `$Pr` will remain cached forever.

chadkirby replied:
Hmm. Good point.

chadkirby committed:
remove removePr method because there's no good use case to remove the $Pr

The removePr method isn't actually used anywhere. I think I was thinking of a use case that probably doesn't exist.

chadkirby committed:
don't cache array properties

because caching the array property breaks the array & it's too fussy to try to maintain the array cache in a valid state (who knew???)
